package/MDAnalysis/analysis/pca.py

```diff
@@ -414,6 +414,42 @@ def transform(self, atomgroup, n_components=None, start=None, stop=None,
 
         return dot
 
+    def project_single_frame(self, components=None):
+        """Runs an inverse-PCA and generates
+           a trajectory that is projected on selected PCs
+
+        Run PCA class before using this function.
+        To project the trajectory onto the first principal component,
+            pca = PCA(universe, select='backbone').run()
+            func = pca.project_single_frame(components=0)
+
+        To apply the projection on-the-fly, may add as a transformation
+            u.trajectory.add_transformations(func)
+
+        Parameters
+        ----------
+        components : int, array, optional
+             Components to be projected onto.
+             The default ``None`` maps onto all components.
+
+        Returns
+        -------
+        Returns the projecting function which transforms a timestep
+        """
+        if components is None:
+            components = range(self.results.p_components.shape[1])
+
+        def wrapped(ts):
+            xyz = self._atoms.positions.ravel() - self._xmean
+            dot = np.dot(xyz, self._p_components[:, components])
+            self._atoms.positions = np.reshape(
+                (np.dot(dot, self._p_components[:, components].T)
+                 + self._xmean), (-1, 3)
```

Comment:
Possibly do everything in one line (better performance?) with [np.linalg.multi_dot](https://numpy.org/doc/stable/reference/generated/numpy.linalg.multi_dot.html).

Also, if we can avoid transposes somehow then that will also speed up — perhaps using [np.einsum](https://numpy.org/doc/stable/reference/generated/numpy.einsum).

Reply:
> Possibly do everything in one line (better performance?) with [np.linalg.multi_dot](https://numpy.org/doc/stable/reference/generated/numpy.linalg.multi_dot.html).
The latest commit does everything in one line. However, there are two issues with [np.linalg.multi_dot](https://numpy.org/doc/stable/reference/generated/numpy.linalg.multi_dot.html). 

1) When the trajectory is projected onto a single component, this multi_dot will fail since it requires the PC component matrix to be 2-D.
2) multi_dot is in fact slower because it tries to figure out the order of dot products by calculating the cost for matrix multiplication. np.dot(np.dot(A, B), C) already optimizes the computation cost because, in our case, A and C are 1-D arguments. 
> Also, if we can avoid transposes somehow then that will also speed up — perhaps using [np.einsum](https://numpy.org/doc/stable/reference/generated/numpy.einsum).

I tested einsum and saw no significant difference in performance. I continue to use self.transpose method. I looked at the inverse-PCA implementation in some other standard codes and they all seem to use the transpose, either in the PCA transformation or in the inverse-PCA transformation, depending on the conventions for matrix axes.

Reply:
Thanks for considering everything, I appreciate your thorough and well-reasoned reply!